be/src/vec/exec/format/orc/vorc_reader.cpp

```diff
@@ -1747,9 +1747,16 @@ Status OrcReader::get_next_block(Block* block, size_t* read_rows, bool* eof) {
             }
         } else {
             if (_delete_rows_filter_ptr) {
+                _execute_filter_position_delete_rowids(*_delete_rows_filter_ptr);
                 RETURN_IF_CATCH_EXCEPTION(Block::filter_block_internal(block, columns_to_filter,
                                                                        (*_delete_rows_filter_ptr)));
+            } else {
+                std::unique_ptr<IColumn::Filter> filter(new IColumn::Filter(block->rows(), 1));
```

Comment:
Why need to create filter when there's no delete rows?

Reply:
_delete_rows_filter_ptr  be used in transactional hive orc , not position delete.